src/fs/metadata_log.cc

```diff
@@ -89,6 +94,11 @@ future<> metadata_log::shutdown() {
     });
 }
 
+void metadata_log::switch_fs_read_only_mode(bool val) noexcept {
+    mlogger.debug("Turned {} read-only mode.", val ? "on" : "off");
```

Comment:
nitpick: It deserves a higher log level, at least `info` - since it's a rather hardcore operation. I'd actually consider issuing a `warning` each time it happens, just so the users know that their system is not fully functional anymore.